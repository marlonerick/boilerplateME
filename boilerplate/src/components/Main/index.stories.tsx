import { Meta } from '@storybook/react/types-7-0'
import Main from './index'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'React AvanÃ§ado',
    description: 'Typescript, ReactJS, NextJS e Styled Components'
  },
  argTypes: {
    BackgroundColor: { control: 'color' }
  }
} as Meta

export const Basic = (
  args: JSX.IntrinsicAttributes & {
    title?: string | undefined
    description?: string | undefined
  }
) => <Main {...args} />

export const Default = (
  args: JSX.IntrinsicAttributes & {
    title?: string | undefined
    description?: string | undefined
  }
) => <Main {...args} />
