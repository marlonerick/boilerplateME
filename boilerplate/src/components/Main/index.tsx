import * as S from './styles'

const Main = ({
  title = 'React Avançado',
  description = 'Typescript, ReactJS, NextJS e Styled Components'
}) => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de um átomo com fundo no formato hexagonal rosa com escrito ao lado React avançado"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>

    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Um desenvolvedor de frente para 3 telas com código."
    ></S.Illustration>
  </S.Wrapper>
)

export default Main
