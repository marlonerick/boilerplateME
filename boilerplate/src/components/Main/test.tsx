import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import Main from '../Main/index'

describe('First test', () => {
  it('should render the heading', async () => {
    const { container } = await render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avanÃ§ado/i })
    ).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle('background-color: #06022b')
    expect(container).toMatchSnapshot()
  })
})
