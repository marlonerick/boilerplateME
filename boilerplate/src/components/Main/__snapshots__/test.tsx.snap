// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First test should render the colors correctly 1`] = `
<div>
  <main
    class="sc-bcXHqe cIRDGJ"
  >
    <img
      alt="Imagem de um átomo com fundo no formato hexagonal rosa com escrito ao lado React avançado"
      class="sc-gswNZR jrWWzK"
      src="/img/logo.svg"
    />
    <h1
      class="sc-dkrFOg cTgUWI"
    >
      React Avançado
    </h1>
    <h2
      class="sc-hLBbgP cYZaHe"
    >
      Typescript, ReactJS, NextJS e Styled Components
    </h2>
    <img
      alt="Um desenvolvedor de frente para 3 telas com código."
      class="sc-eDvSVe hbgsoG"
      src="/img/hero-illustration.svg"
    />
  </main>
</div>
`;

exports[`First test should render the heading 1`] = `
<div>
  <main
    class="sc-bcXHqe cIRDGJ"
  >
    <img
      alt="Imagem de um átomo com fundo no formato hexagonal rosa com escrito ao lado React avançado"
      class="sc-gswNZR jrWWzK"
      src="/img/logo.svg"
    />
    <h1
      class="sc-dkrFOg cTgUWI"
    >
      React Avançado
    </h1>
    <h2
      class="sc-hLBbgP cYZaHe"
    >
      Typescript, ReactJS, NextJS e Styled Components
    </h2>
    <img
      alt="Um desenvolvedor de frente para 3 telas com código."
      class="sc-eDvSVe hbgsoG"
      src="/img/hero-illustration.svg"
    />
  </main>
</div>
`;
